src/python/pants/backend/jvm/tasks/coursier_resolve.py

```diff
@@ -0,0 +1,478 @@
+
+  def _write_result_to_cache(self, compile_classpath, all_vts, target_resolution_filename):
+    # TODO(wisechengyi): currently the path contains abs path, so need to remove that before
+    # cache can be shared across machines.
+    for vt in all_vts:
+      t = vt.target
+      if isinstance(t, JarLibrary):
+        with open(os.path.join(vt.results_dir, target_resolution_filename), 'w') as f:
+          pickle.dump(compile_classpath.get_artifact_classpath_entries_for_targets([t]), f)
+        vt.update()
+
+  def _load_result_from_cache(self, compile_classpath, all_vts, target_resolution_filename):
+    """
+
+    :param compile_classpath:
+    :param all_vts:
+    :param target_resolution_filename:
+    :return: True if success; False if any of the classpath is not valid anymore.
+    """
+    temp_store = []
+
+    for vt in all_vts:
+      t = vt.target
+      if isinstance(t, JarLibrary):
+        # compile_classpath.add_jars_for_targets
+        with open(os.path.join(vt.results_dir, target_resolution_filename), 'r') as f:
+          tuples_conf_artifact_classpath = pickle.load(f)
+          for conf, artifact_classpath in tuples_conf_artifact_classpath:
+            if not os.path.exists(artifact_classpath.path) \
+                or not os.path.exists(artifact_classpath.cache_path) \
+                or not os.path.exists(os.path.realpath(artifact_classpath.cache_path)):
+              print("cache verification failed")
+              return False
+            temp_store.append((t, tuples_conf_artifact_classpath))
+
+    # If all artifacts path are valid, add them to compile_classpath
+    for t, tuples_conf_artifact_classpath in temp_store:
+      compile_classpath.add_elements_for_target(t, tuples_conf_artifact_classpath)
+    print("cache verified")
+    return True
+
+  def _construct_cmd_args(self, classified_jars, classifier, common_args, global_excludes, local_exclude_args,
+                          pinned_coords, coursier_workdir):
+    cmd_args = list(common_args)
+    if classifier:
+      cmd_args.extend(['--classifier', classifier])
+    for j in classified_jars:
+      if not j.rev:
+        continue
+
+      if j.intransitive:
+        cmd_args.append('--intransitive')
+
+      cmd_args.append(j.coordinate.simple_coord)
+
+      # Force requires specifying the coord again with -V
+      if j.force:
+        cmd_args.append('-V')
+        cmd_args.append(j.coordinate.simple_coord)
+
+    # Force pinned coordinates
+    for j in pinned_coords:
+      cmd_args.append('-V')
+      cmd_args.append(j.simple_coord)
+
+    if local_exclude_args:
+
+      with temporary_file(coursier_workdir, cleanup=False) as f:
+        exclude_file = f.name
+        with open(exclude_file, 'w') as ex_f:
+          ex_f.write('\n'.join(local_exclude_args).encode('utf8'))
+
+        cmd_args.append('--soft-exclude-file')
+        cmd_args.append(exclude_file)
+
+    # TODO(wisechengyi): support exclusion on the whole org
```

Comment:
You could make it an error, if you're planning on deferring it for now.

Reply:
figured out the syntax to pass to coursier. `-E org:*`